argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,274 @@
+# <Copyright 2020, Argo AI, LLC. Released under the MIT license.>
+import argparse
+import logging
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from multiprocessing import Pool
+from pathlib import Path
+from typing import List, Tuple
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+from argoverse.data_loading.object_label_record import read_label
+from argoverse.evaluation.detection_utils import (
+    DistFnType,
+    SimFnType,
+    compute_match_matrix,
+    dist_fn,
+    filter_instances,
+    get_ranks,
+    interp,
+)
+
+matplotlib.use("Agg")
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class DetectionCfg:
+    """Instantiates a DetectionCfg object for configuring a DetectionEvaluator.
+
+    Args:
+        sim_ths: The similarity thresholds for determining a true positive.
+        sim_fn_type: The type of similarity function to be used for calculating
+                     average precision.
+        n_rec_samples: Number of recall points to sample uniformly in [0, 1].
+        dcs: The weight vector for the detection composite score (DCS).
+        tp_threshold: Center distance threshold for the true positive metrics.
+        significant_digits: The precision for metrics.
+    """
+
+    sim_ths: Tuple[float] = (0.5, 1.0, 2.0, 4.0)
+    sim_fn_type: SimFnType = SimFnType.CENTER
+    n_rec_samples: int = 101  # 101 gives us one sample at ever integer from 0 to 100
+    cds_weights: Tuple[float] = (3, 1, 1, 1)
```

Comment:
can we add an inline comment above explaining which metrics 3, 1, 1, 1 correspond to?